src/testIntegration/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/security/SmoketestTest.kt

```diff
@@ -0,0 +1,104 @@
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = null))
+        .body(BodyInserters.fromValue("""{ "secret": "some-lsj-secret" }"""))
+        .exchange()
+        .expectStatus().isCreated
+        .returnResult(VerifyLinkResponse::class.java)
+        .responseBody
+        .blockFirst()
+        ?: Assertions.fail("Did not receive a response from /link/verify")
+
+      assertThat(jwtService.validateToken(jwt.token)).isTrue
+      assertThat(jwtService.subject(jwt.token)).isEqualTo("smoke-test-lsj")
+      assertThat(jwtService.authorities(jwt.token)).containsExactly("ROLE_SLM_CREATE_BARCODE")
+    }
+
+    @Test
+    fun `A mailroom staff smoke test can check barcodes without an auth error`() {
+      val barcode = barcodeRepository.save(Barcode(code = "some-barcode"))
+      barcodeEventRepository.save(BarcodeEvent(barcode = barcode, userId = "any-user", eventType = BarcodeEventType.CREATED, ipAddress = "127.0.0.1"))
+
+      webTestClient.post()
+        .uri("/barcode/check")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = "some-msj-secret", roles = listOf("ROLE_SLM_SCAN_BARCODE")))
+        .body(BodyInserters.fromValue("""{ "barcode": "some-barcode" }"""))
+        .exchange()
+        .expectStatus().isOk
+
+      val checkedEvent = barcodeEventRepository.findByBarcodeAndEventTypeOrderByCreatedDateTime(barcode, BarcodeEventType.CHECKED).first()
+
+      assertThat(checkedEvent.location).isEqualTo("SKI") // LEI is stubbed by Auth but SKI is hardcoded for smoke tests
+    }
+  }
+
+  @Nested
+  inner class SmokeTestDisallowed {
+
+    @Test
+    fun `A legal sender smoke test cannot sign in using the back door`() {
+      webTestClient.post()
+        .uri("/link/verify")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = null))
+        .body(BodyInserters.fromValue("""{ "secret": "some-lsj-secret" }"""))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `A mailroom staff smoke test checking barcodes receives an error because the user doesn't exist`() {
+      HmppsAuthExtension.hmppsAuthApi.stubFailToGetUserDetails()
+      val barcode = barcodeRepository.save(Barcode(code = "no-smoketest"))
+      barcodeEventRepository.save(BarcodeEvent(barcode = barcode, userId = "any-user", eventType = BarcodeEventType.CREATED, ipAddress = "127.0.0.1"))
+
+      webTestClient.post()
+        .uri("/barcode/check")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .headers(setAuthorisation(user = "some-msj-user", roles = listOf("ROLE_SLM_SCAN_BARCODE")))
+        .body(BodyInserters.fromValue("""{ "barcode": "no-smoketest" }"""))
+        .exchange()
+        .expectStatus().is5xxServerError
```

Comment:
This is wrong - we should either return a 404 or maybe a 401. To do this we need to catch the error returned from calling hmpps auth `/api/user/me` but I can't make that call fail right now - it takes the user from the token and I can't generate a token for a user that doesn't exist 😆 

So once this is in dev I can make it break and then work out how to return a 401 or 404.